src/explore-education-statistics-common/src/components/CopyLinkButton.tsx

```diff
@@ -1,51 +1,80 @@
 import Button from '@common/components/Button';
-import LinkIcon from '@common/components/LinkIcon';
 import styles from '@common/components/CopyLinkButton.module.scss';
-import Modal from '@common/components/Modal';
+import { CopyLinkModalProps } from '@common/components/CopyLinkModal';
 import UrlContainer from '@common/components/UrlContainer';
-import VisuallyHidden from '@common/components/VisuallyHidden';
 import useToggle from '@common/hooks/useToggle';
 import classNames from 'classnames';
-import React from 'react';
+import { OmitStrict } from '@common/types';
+import React, { ReactNode } from 'react';
 
-interface Props {
-  className?: string;
-  url: string;
+interface Props extends OmitStrict<CopyLinkModalProps, 'buttonClassName'> {
```

Comment:
This isn't very clean and it'll create an unnecessary circular dependency with `CopyLinkModal`.

Would suggest moving the props from `CopyLinkModal` to here instead and then extending `CopyLinkButtonProps` in `CopyLinkModal` i.e.

```ts
// CopyLinkButton.tsx

export interface CopyLinkButtonProps {
  buttonClassName?: string;
  className?: string;
  confirmMessage?: string;
  inlineButton?: boolean;
  url: string;
}

// CopyLinkModal.tsx

interface CopyLinkModalProps extends CopyLinkButtonProps {
  // ...
}
```